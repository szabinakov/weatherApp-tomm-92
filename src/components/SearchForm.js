import React from "react";
import "../styles/SearchForm.css";

function SearchForm({ searchText, setSearchText, onSubmit }) {
  const handleInputChange = (event) => setSearchText(event.target.value);

  return (
    <div className="search-form">
      <input
        className="search-bar"
        type="text"
        placeholder="Search..."
        onChange={handleInputChange}
        value={searchText}
      />
      <br />
      <button
        type="submit"
        onClick={onSubmit}
        className="search-form__submit-button"
      >
        Search
      </button>
    </div>
  );
}

export default SearchForm;
