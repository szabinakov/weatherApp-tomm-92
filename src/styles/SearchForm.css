.search-form {
  text-align: center;
  margin-top: 60px;
}

.search-bar {
  min-width: 300px;
}
.search-form__submit-button {
  margin-top: 10px;
  font-size: large;
  font-weight: bold;
}

.search-form__submit-button:hover {
  color: red;
}
