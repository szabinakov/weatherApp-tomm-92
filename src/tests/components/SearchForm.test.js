import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "Liverpool",
    setSearchText: () => {},
    onSubmit: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("render the correct text on the submit button", () => {
    const { getByText } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(getByText("Search")).toHaveClass("search-form__submit-button");
  });
});
