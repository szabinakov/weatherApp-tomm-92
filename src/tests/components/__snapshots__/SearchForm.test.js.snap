// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      class="search-bar"
      placeholder="Search..."
      type="text"
      value="Liverpool"
    />
    <br />
    <button
      class="search-form__submit-button"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
